From 66271f0936eb591efef44b09e50a245687882cc8 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 4 Mar 2014 11:55:43 -0800
Subject: [PATCH] drm: xilinx: plane: Remove an unused arg of
 xilinx_drm_plane_mode_set()

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

In xilinx_drm_plane_mode_set(), struct drm_crtc is never used, so
the argument is removed from the function prototype.

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit f65dfeb29e3e6f415905e90f4c17c2494e88654e)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
Signed-off-by: Weiwei Wang <weiwei.wang@windriver.com>

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_crtc.c b/drivers/gpu/drm/xilinx/xilinx_drm_crtc.c
index 66059dd5b76d..dd89cf3d7acf 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_crtc.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_crtc.c
@@ -164,7 +164,7 @@ static int xilinx_drm_crtc_mode_set(struct drm_crtc *base_crtc,
 					 adjusted_mode->vdisplay);
 
 	/* configure a plane: vdma and osd layer */
-	ret = xilinx_drm_plane_mode_set(crtc->priv_plane, base_crtc,
+	ret = xilinx_drm_plane_mode_set(crtc->priv_plane,
 					base_crtc->primary->fb, 0, 0,
 					adjusted_mode->hdisplay,
 					adjusted_mode->vdisplay,
@@ -187,7 +187,7 @@ static int _xilinx_drm_crtc_mode_set_base(struct drm_crtc *base_crtc,
 	int ret;
 
 	/* configure a plane */
-	ret = xilinx_drm_plane_mode_set(crtc->priv_plane, base_crtc,
+	ret = xilinx_drm_plane_mode_set(crtc->priv_plane,
 					fb, 0, 0,
 					base_crtc->hwmode.hdisplay,
 					base_crtc->hwmode.vdisplay,
diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_plane.c b/drivers/gpu/drm/xilinx/xilinx_drm_plane.c
index 0c371ffd2f47..78b2ba193bdc 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_plane.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_plane.c
@@ -225,7 +225,7 @@ void xilinx_drm_plane_commit(struct drm_plane *base_plane)
 
 /* mode set a plane */
 int xilinx_drm_plane_mode_set(struct drm_plane *base_plane,
-			      struct drm_crtc *crtc, struct drm_framebuffer *fb,
+			      struct drm_framebuffer *fb,
 			      int crtc_x, int crtc_y,
 			      unsigned int crtc_w, unsigned int crtc_h,
 			      uint32_t src_x, uint32_t src_y,
@@ -303,7 +303,7 @@ static int xilinx_drm_plane_update(struct drm_plane *base_plane,
 {
 	int ret;
 
-	ret = xilinx_drm_plane_mode_set(base_plane, crtc, fb,
+	ret = xilinx_drm_plane_mode_set(base_plane, fb,
 					crtc_x, crtc_y, crtc_w, crtc_h,
 					src_x >> 16, src_y >> 16,
 					src_w >> 16, src_h >> 16);
diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_plane.h b/drivers/gpu/drm/xilinx/xilinx_drm_plane.h
index ec9301a09d46..a262de55d60b 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_plane.h
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_plane.h
@@ -25,7 +25,7 @@ struct drm_plane;
 void xilinx_drm_plane_dpms(struct drm_plane *base_plane, int dpms);
 void xilinx_drm_plane_commit(struct drm_plane *base_plane);
 int xilinx_drm_plane_mode_set(struct drm_plane *base_plane,
-			      struct drm_crtc *crtc, struct drm_framebuffer *fb,
+			      struct drm_framebuffer *fb,
 			      int crtc_x, int crtc_y,
 			      unsigned int crtc_w, unsigned int crtc_h,
 			      uint32_t src_x, uint32_t src_y,
-- 
2.1.0

